wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+    and gets the mean and standard  deviation of this gaussian: `f_m` and `f_std`.
+    
+    Then, fits `h` (the unfiltered charge histogram) with a gaussian around `f_m`,
+    and gets the mean, standard deviation, and scale of this gaussian: `m`, `std`, `scale`.
+
+    Returns `(m, f_std, std, scale)`
+
+    If `f_h` is passed as `None`, only default values are returned, specified as global constants
+    in this module (except for the default scale, which is proportional to the entries in `h`).
+    """
+    if f_h == None:
+        return (D_OFFSET, D_NOISE_SPREAD, D_ZERO_PEAK_SPREAD, h.GetEntries()*0.075)
+    else:
+        filter_fit = ROOT.TF1('filter_fit', 'gaus', f_h.GetMean() - 2*f_h.GetStdDev(), f_h.GetMean() + 2*f_h.GetStdDev())
+        f_h.Fit(filter_fit, 'SRQ0')
+        offset = filter_fit.GetParameter(1)
+        noise_spread = filter_fit.GetParameter(2)
+        
+        win = max(0.3, 20*noise_spread)
+        raw_fit = ROOT.TF1('raw_fit', 'gaus', offset - win, offset + win)
+        raw_fit.SetParameter(1, offset)
+        for _ in range(3):
+            h.Fit(raw_fit, 'SRQ')
+            offset = raw_fit.GetParameter(1)
+            raw_fit.SetRange(offset-win, offset+win)
+        raw_spread = raw_fit.GetParameter(2)
+        scale = raw_fit.GetParameter(0)
+        return (offset, noise_spread, raw_spread, scale)
+
+def peaks(h, width=4, height=0.05):
+    """
+    Finds peaks in hisogram `h`.
+    `height` is measured as a fraction of the highest peak.
+    Peaks lower than `<highest peak>*height` will not be searched for.
+    """
+    spec = ROOT.TSpectrum()
+    highest_peak = None
+    n_pks = spec.Search(h, width, "nobackground", height)               
+    x_pos = spec.GetPositionX()
+    x_pos = np.array([x_pos[i] for i in range(n_pks)])
+    if len(x_pos) != 0:
+        highest_peak = x_pos[0]
+    ind = np.argsort(x_pos)
+    x_pos = x_pos[ind]
+    return (x_pos, highest_peak)
+    # print(f'All peaks: {x_pos}')
+    # x_pos = remove_bad_peaks(h, x_pos, 0.5)
+    # print(f'Good peaks: {x_pos}')
+
+def get_bin_num(h, val):
+    """
+    Gets the bin number of `val` if `val` was placed into the histogram `h`.
+    Does NOT add `val` to `h`.
+    """
+    return h.GetXaxis().FindBin(val)
+
+class poisson_model:
+    """
+    p[0]: overall scale
+    p[1]: offset
+    p[2]: mean number of SPEs in integration window
+    p[3]: SPE charge
+    p[4]: std. of scope/digitizer noise
+    p[5]: std. of SPE charge
+    """
+    def __call__(self, x, p):
+        # Original model in root syntax:
+        # string = "[0]*(" + "+".join(["TMath::Poisson(%i,[2])*TMath::Gaus(x-[1],[3]*%i,TMath::Sqrt([4]^2+[5]^2*%i))" % (i,i,i) for i in range(0,n)]) + ")"
+        
+        model = 0
+        for i in range(n):
+            model += TMath.Poisson(i, p[2]) * TMath.Gaus(x[0]-p[1], i*p[3], TMath.Sqrt(p[4]**2 + i*(p[5]**2)), False)
+        model *= p[0]
+        return model
+
+def fac(x):
+    """
+    factorial
```

Comment:
I would change this to:

```
"""
Returns x!. We use the gamma function here instead since it works even for non-integer values and is generally faster than calling math.factorial(x)
```